src/linter/root.go

```diff
@@ -857,11 +857,14 @@ func (d *RootWalker) enterClassConstList(s *stmt.ClassConstList) bool {
 		d.checkCommentMisspellings(c, c.PhpDocComment)
 		typ := solver.ExprTypeLocal(d.scope(), d.ctx.st, c.Expr)
 
+		value, _ := solver.GetConstantValue(c, d.ctx.st)
```

Comment:
GetConstantValue may call GetConstant and this creates a bad situation where the order of the file traversal matters. We try to avoid that.

In general, you can't use solver methods that rely on the metadata during the indexing.

I would limit GetConstantValue to AST nodes and avoid querying the metadata during the indexing.